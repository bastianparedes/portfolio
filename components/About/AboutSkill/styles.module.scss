.skillCard {
  width: 5rem;
  position: relative;
}

.skillImageContainer {
  width: 100%;
  height: 5rem;
  position: relative;
  margin-bottom: 0.5rem;
}

.skillName {
  font-weight: bolder;
  text-align: center;
  margin: 0;
}

@media screen and (max-width: 768px) {
  .skillCard {
    width: 5rem;
  }
  .skillName {
    font-size: 1rem;
  }
}

@media screen and (max-width: 480px) {
  .skillCard {
    width: 4rem;
  }
  .skillName {
    font-size: 0.8rem;
  }
}
