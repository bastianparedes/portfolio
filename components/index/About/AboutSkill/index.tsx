import React from 'react';

import Image from 'next/legacy/image';

import styles from './styles.module.scss';

interface SkillProps {
  src: string;
  name: string;
}

const AboutSkill = (props: SkillProps): JSX.Element => {
  return (
    <div className={styles.skillCard}>
      <div className={styles.skillImageContainer}>
        <Image
          alt={props.name}
          layout="fill"
          objectFit="contain"
          placeholder="blur"
          src={'/images/skills/' + props.src}
        />
      </div>
      <p className={styles.skillName}>{props.name}</p>
    </div>
  );
};

export default AboutSkill;
