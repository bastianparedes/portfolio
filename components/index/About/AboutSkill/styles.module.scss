@import 'styles/_breakpoints';

.skillCard {
  width: 5rem;
  position: relative;
}

.skillImageContainer {
  width: 100%;
  height: 5rem;
  position: relative;
  margin-bottom: 0.5rem;
}

.skillName {
  font-weight: bolder;
  text-align: center;
  margin: 0;
}

@include phone-and-tablet {
  .skillCard {
    width: 5rem;
  }
  .skillName {
    font-size: 1rem;
  }
}

@include phone {
  .skillCard {
    width: 4rem;
  }
  .skillName {
    font-size: 0.8rem;
  }
}
