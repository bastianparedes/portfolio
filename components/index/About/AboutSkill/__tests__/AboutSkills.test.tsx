import React from 'react';
import AboutSkill from '..';
import { render } from '@testing-library/react';

describe('<AboutSkill />', () => {
  const skill = {
    SRC: 'html5.png',
    NAME: 'HTML 5'
  };
  const props = {
    src: skill.SRC,
    name: skill.NAME
  };
  it('should render', () => {
    const { container } = render(<AboutSkill {...props} />);
    expect(container).toMatchSnapshot();
  });
});
