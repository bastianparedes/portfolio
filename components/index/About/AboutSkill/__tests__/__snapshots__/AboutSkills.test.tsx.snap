// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutSkill /> should render 1`] = `
<div>
  <div
    class="skillCard"
  >
    <div
      class="skillImageContainer"
    />
    <p
      class="skillName"
    >
      HTML 5
    </p>
  </div>
</div>
`;
